examples/binary_segmentation_buildings.py

```diff
@@ -0,0 +1,532 @@
+import logging
+import os
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+from torch.optim import lr_scheduler
+from torch.utils.data import DataLoader
+from torch.utils.data import Dataset as BaseDataset
+from tqdm import tqdm
+
+import segmentation_models_pytorch as smp
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(message)s",
+    datefmt="%d:%m:%Y %H:%M:%S",
+)
+
+
+"""
+This script demonstrates how to train a binary segmentation model using the 
+CamVid dataset and segmentation_models_pytorch. The CamVid dataset is a 
+collection of videos with pixel-level annotations for semantic segmentation.
+The dataset includes 367 training images, 101 validation images, and 233 test.
+Each training image has a corresponding mask that labels each pixel as belonging
+to these classes with the numerical labels as follows:
+- Sky: 0
+- Building: 1
+- Pole: 2
+- Road: 3
+- Pavement: 4
+- Tree: 5
+- SignSymbol: 6
+- Fence: 7
+- Car: 8
+- Pedestrian: 9
+- Bicyclist: 10
+- Unlabelled: 11
+ 
+In this script, we focus on binary segmentation, where the goal is to classify 
+each pixel as whether belonging to a certain class (Foregorund) or 
+not (Background).
+
+Class Labels:
+- 0: Background
+- 1: Foreground
+
+The script includes the following steps:
+1. Set the device to GPU if available, otherwise use CPU.
+2. Download the CamVid dataset if it is not already present.
+3. Define hyperparameters for training.
+4. Define a custom dataset class for loading and preprocessing the CamVid 
+     dataset.
+5. Define a function to visualize images and masks.
+6. Create datasets and dataloaders for training, validation, and testing.
+7. Define a model class for the segmentation task.
+8. Train the model using the training and validation datasets.
+9. Evaluate the model using the test dataset and save the output masks and 
+     metrics.
+"""
+
+# ----------------------------
+# Set the device to GPU if available
+# ----------------------------
+if torch.cuda.is_available():
+    device = torch.device("cuda")
+    logging.info("Using GPU")
+else:
+    device = torch.device("cpu")
+    logging.info("Using CPU")
+    os.system("export OMP_NUM_THREADS=64")
+    torch.set_num_threads(os.cpu_count())
```

Comment:
A bit more common way

```suggestion
device = "cuda" if torch.cuda.is_available() else "cpu"
logging.info(f"Using device: {device}")

if device == "cpu":
    os.system("export OMP_NUM_THREADS=64")
    torch.set_num_threads(os.cpu_count())
```